### PR TITLE
Hip17 Unhappy Account State Tests

### LINKED ISSUE #1788: Create test matrix for HIP-17 operations vs. unhappy account states
:bangbang: This issue should be broken down into sub-issues, maybe by HAPI operation?

### Problem

HIP-17 affects all HTS operations (including three new queries), as well as `CryptoTransfer`.

In most cases these operations involve accounts; and besides the "happy" state of a non-expired, un-frozen account that is associated to all the necessary tokens with KYC granted, an account may be:
- [ ] [Dissociated from a relevant token](https://github.com/hashgraph/hedera-services/issues/1804).
- [ ] [Frozen for a relevant token](https://github.com/hashgraph/hedera-services/issues/1805).
- [ ] [Without KYC for a relevant token](https://github.com/hashgraph/hedera-services/issues/1806).
- [ ] [Expired](https://github.com/hashgraph/hedera-services/issues/1807).
- [ ] [Deleted](https://github.com/hashgraph/hedera-services/issues/1808).
- [ ] [Auto-removed](https://github.com/hashgraph/hedera-services/issues/1809).

We need to validate (and characterize via a regression test) the behavior of each operation when it faces an account in a non-happy state.

### Solution

Create a devoted `Hip17UnhappyAccountsSuite` with a spec characterizing how each HIP-17 operation responds to each case.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by anighanta**
**Description**:

Adds tests for unhappy account state scenarios when operating on/with a NFT

**Related issue(s)**:

Fixes #1788 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by anighanta** [comment]
Will be finalized when PR #1877 is merged

File: hedera-node/src/test/java/com/hedera/services/state/expiry/renewal/RenewalHelperTest.java [hunk 1]
```diff
@@ -123,13 +124,15 @@
 	@Mock
 	private FCMap<MerkleEntityAssociation, MerkleTokenRelStatus> tokenRels;
 	@Mock
+	private BackingAccounts backingAccounts;
```
Comment (abhishek-hedera) [inline hedera-node/src/test/java/com/hedera/services/state/expiry/renewal/RenewalHelperTest.java:127]: Shouldn't this be of type `BackingStore`?

Comment (anighanta) [reply hedera-node/src/test/java/com/hedera/services/state/expiry/renewal/RenewalHelperTest.java]: This change got merged with another PR #1877. Yes we ca use BackingStore as well. But in particular as we are using https://github.com/hashgraph/hedera-services/blob/cb9110ddba5ad6b6abd2b60c6be469f23679e86a/hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java#L1655

We will always get a `BackingAccounts`

Comment (abhishek-hedera) [reply hedera-node/src/test/java/com/hedera/services/state/expiry/renewal/RenewalHelperTest.java]: I see, thanks! Looks good overall.

**Review by abhishek-hedera — APPROVED** [review]
LGTM

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyAccountsSuite.java [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+		});
+	}
+
+	private HapiApiSpec uniqueTokenOperationsFailForAutoRemovedAccount() {
+		return defaultHapiSpec("UniqueTokenOperationsFailForAutoRemovedAccount")
+				.given(
+						fileUpdate(APP_PROPERTIES)
+								.payingWith(GENESIS)
+								.overridingProps(Map.of(
+										"tokens.nfts.areEnabled", "true",
```
Comment (ljianghedera) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyAccountsSuite.java:99]: Is this by default enabled now?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyAccountsSuite.java [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+										"ledger.autoRenewPeriod.minDuration", "1",
+										"autorenew.isEnabled", "true",
+										"autorenew.gracePeriod", "0",
+										"autorenew.numberOfEntitiesToScan", "100",
+										"autorenew.maxNumberOfEntitiesToRenewOrDelete", "10"
```
Comment (ljianghedera) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyAccountsSuite.java:104]: nit: better add the `erasingProps(Set.of("minimumAutoRenewDuration"))` here

**Review by ljianghedera — DISMISSED** [review]
Overall LGTM.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyAccountsSuite.java (hunk 2 above)
Comment (anighanta) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyAccountsSuite.java:99]: will remove

**Review by abhishek-hedera — COMMENTED** [review]
Merge conflicts need to be resolved before merging. LGTM

**Review by abhishek-hedera — APPROVED** [review]

**Pull request merged**